cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -440,6 +462,12 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
           out_buffers.emplace_back(static_cast<size_type*>(out_buf.data()) + out_buf.size);
           final_offsets.emplace_back(static_cast<size_type>(col_string_sizes[idx]));
         }
+        // Nested large strings column
+        else if (input_col.nesting_depth() > 0) {
+          CUDF_EXPECTS(static_cast<int64_t>(initial_str_offsets[idx]) >= int64_t{0},
+                       "Initial string offset must be >= 0");
```

Comment:
I don't understand this check. How can `initial_str_offsets[idx]`, which is size_t, be negative?

Reply:
omg, is this because we initialize to `std::numeric_limits<size_t>::max()`?

Reply:
Yes exactly, as the decoder should set it to `>0` for all large-strings cols!